tendrl/integrations/ceph/sds_sync/__init__.py

```diff
@@ -0,0 +1,95 @@
+from tendrl.node_agent.node_sync import disk_sync
+
+
+class CephIntegrtaionsSyncThread(sds_sync.StateSyncThread):
+    def _run(self):
+        Event(
+            Message(
+                priority="info",
+                publisher=NS.publisher_id,
+                payload={"message": "%s running" % self.__class__.__name__}
+            )
+        )
+        while not self._complete.is_set():
+            # sync of OSD journal details every 30 sec is fine
+            # as this data doesnt change very frequently in cluster
+            gevent.sleep(30)
+            try:
+                # Sync the OSD journal details
+                Event(
+                    Message(
+                        priority="info",
+                        publisher=NS.publisher_id,
+                        payload={
+                            "message":
+                            "ceph_integrations_sync, osd journal details"
+                        }
+                    )
+                )
+                journal_details = {}
+                osd_list = os.listdir("/var/lib/ceph/osd")
+                if len(osd_list) > 0:
+                    for item in os.listdir("/var/lib/ceph/osd"):
+                        journal_path = "/var/lib/ceph/osd/%s/journal" % item
+                        out, err, rc = cmd_utils.Command(
+                            "lsblk -o PKNAME,ROTA -n %s" % journal_path
+                        ).run()
+                        if rc != 0:
+                            Event(
+                                Message(
+                                    priority="error",
+                                    publisher=NS.publisher_id,
+                                    payload={
+                                        "message":
+                                        "Error getting journal details for OSD. Error: %s" %
+                                        err
+                                    }
+                                )
+                            )
+                        else:
+                            journal_disk, rotational = out.split()
+                            if ("/dev/%s" % journal_disk) in journal_details.keys():
+                                journal_details[
+                                    "/dev/%s" % journal_disk
+                                ]['journal_count'] += 1
+                                journal_details[
+                                    "/dev/%s" % journal_disk
+                                ]['ssd'] = disk_sync.is_ssd(rotational)
+                            else:
+                                journal_details["/dev/%s" % journal_disk] = {
+                                    'journal_count': 1,
+                                    'ssd': disk_sync.is_ssd(rotational)
+                                }
+
+                    NS.integrations.ceph.objects.Journal(
+                        integration_id=NS.tendrl_context.integration_id,
```

Comment:
Thats a bad idea

This entire sync while loop should run only if NS.tendrl_context.integration_id is not None, this is the "integrations.$sds" namespace because it will require an integration id always